### PR TITLE
add LogUpkeepCounter to chaincli

### PR DESCRIPTION
**Pull request by RyanRHall**

### CODE REVIEW

**Review by ferglor — DISMISSED** [review]
Just one comment but looks cool!

File: core/scripts/chaincli/handler/keeper.go [hunk 1]
```diff
@@ -431,23 +453,67 @@ func (k *Keeper) deployUpkeeps(ctx context.Context, registryAddr common.Address,
...
+				"topic0":          "0x3d53a39550e04688065827f3bb86584cb007ab9ebca7ebd528e7301c9c31eb5d", // event sig for Trigger()
+				"topic1":          "0x",
+				"topic2":          "0x",
+				"topic3":          "0x",
+			}, logTriggerConfigType)
+			if err1 != nil {
+				log.Fatal("error here 1", err1)
+			}
+			var err2 error
+			extraData, err2 = extraDataEncoder.Encode([]interface{}{1, logTriggerConfig, "0x"})
```
Comment (shileiwill) [inline core/scripts/chaincli/handler/keeper.go:513]: nit: `extraData, err2 := extraDataEncoder.Encode([]interface{}{1, logTriggerConfig, "0x"})` could be neater. I guess you can reuse the err variable for the above err1 and err2 as well?

File: core/scripts/chaincli/handler/keeper.go [hunk 2]
```diff
@@ -473,27 +539,36 @@ func (k *Keeper) deployUpkeeps(ctx context.Context, registryAddr common.Address,
 
 	var err error
 	var upkeepGetter activeUpkeepGetter
+	var endIdx *big.Int // second arg in getActiveUpkeepIds (breaking change in v2.1)
```
Comment (shileiwill) [inline core/scripts/chaincli/handler/keeper.go:542]: this seems to be the 4th arg in getActiveUpkeepIds as shown in `activeUpkeepIds := k.getActiveUpkeepIds(ctx, upkeepGetter, big.NewInt(existingCount), endIdx)`

it took me some time to figure out this is endIdx or count, and this is inclusive or not. Seems we are changing from "count" to "index", you may want to update the function signature of `GetActiveUpkeepIDs(opts *bind.CallOpts, startIndex *big.Int, maxCount *big.Int)` in `type activeUpkeepGetter` as well.

**Review by shileiwill — APPROVED** [review]

File: contracts/src/v0.8/automation/tests/LogUpkeepCounter.sol [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+  constructor(uint256 _testRange) {
+    testRange = _testRange;
+    previousPerformBlock = 0;
+    lastBlock = block.number;
+    initialBlock = 0;
+    counter = 0;
+  }
+
+  function start() public {
```
Comment (amirylm) [inline contracts/src/v0.8/automation/tests/LogUpkeepCounter.sol:36]: Good coverage of existing scenarios, but we might need additional function that accepts the parameters to emit Trigger, then we could perform any case from outside the contract

**Review by amirylm — APPROVED** [review]

**Comment by amirylm** [comment]
Missing teardown for 2.1 in the end of `LaunchAndTest` to cancel/withdraw upkeeps

**Review by amirylm — COMMENTED** [review]
Missing teardown for 2.1 at the end of `LaunchAndTest` (cancel/withdraw active upkeeps)

File: core/scripts/chaincli/handler/keeper.go (hunk 2 above)
Comment (RyanRHall) [reply core/scripts/chaincli/handler/keeper.go:542]: sorry for the confusion. I meant 2nd arg on registry function :man_facepalming: .

Comment (RyanRHall) [reply core/scripts/chaincli/handler/keeper.go:542]: added a bunch of additional comments to hopefully make it less confusing

File: contracts/src/v0.8/automation/tests/LogUpkeepCounter.sol (hunk 3 above)
Comment (RyanRHall) [reply contracts/src/v0.8/automation/tests/LogUpkeepCounter.sol:36]: I will save that for a later PR when we have more concrete demands. Hopefully this is good enough for now :)

**Review by shileiwill — APPROVED** [review]

**Review by shileiwill — APPROVED** [review]

**Pull request merged**